api/v1alpha1/project_config_types.go

```diff
@@ -342,6 +345,147 @@ type AzureWebhookReceiverConfig struct {
+	//
+	// +kubebuilder:validation:MinItems=1
+	Targets []GenericWebhookTarget `json:"targets,omitempty" protobuf:"bytes,4,rep,name=targets"`
+}
+
+// GenericWebhookActionName represents the name of an action to be performed on a resource.
+type GenericWebhookActionName string
+
+const (
+	// GenericWebhookActionNameRefresh indicates a request to refresh the resource.
+	GenericWebhookActionNameRefresh GenericWebhookActionName = "Refresh"
+)
+
+// GenericWebhookTarget describes a target resource for a generic webhook event.
+type GenericWebhookTarget struct {
+	// Kind is the kind of the target resource (e.g., "Warehouse", "Stage").
+	//
+	// +kubebuilder:validation:Required
+	Kind string `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+
+	// LabelSelector is a label selector to identify the target resources.
+	//
+	// +optional
+	LabelSelector metav1.LabelSelector `json:"labelSelector,omitempty" protobuf:"bytes,2,opt,name=labelSelector"`
+
+	// IndexSelector is an index selector to identify the target resources.
+	//
+	// +optional
+	IndexSelector IndexSelector `json:"indexSelector,omitempty" protobuf:"bytes,3,opt,name=indexSelector"`
+}
+
+// IndexSelector encapsulates a selector used to derive index keys
+// based on expressions.
+type IndexSelector struct {
+	// MatchExpressions is a list of index selector requirements.
+	//
+	// +kubebuilder:validation:MinItems=1
+	MatchExpressions []IndexSelectorRequirement `json:"matchExpressions,omitempty" protobuf:"bytes,1,rep,name=matchExpressions"`
+}
+
+// IndexSelectorRequirement encapsulates a requirement used to select indexes
+// based on specific criteria.
+type IndexSelectorRequirement struct {
+	// Key is the key of the index.
+	//
+	// kubebuilder:validation:Required
+	Key string `json:"key" protobuf:"bytes,1,opt,name=key"`
+
+	// Operator indicates the operation that should be used to evaluate
+	// whether the selection requirement is satisfied.
+	//
+	// kubebuilder:validation:Required
+	Operator Operator `json:"operator" protobuf:"bytes,2,opt,name=operator"`
+
+	// Values is a list of values or a single value returned from an expression.
+	//
+	// kubebuilder:validation:Required
+	Values apiextensionsv1.JSON `json:"values" protobuf:"bytes,3,opt,name=values"`
+}
+
+// ConditionSelector encapsulates a condition used to match resources based
+// on specific criteria.
+type ConditionSelector struct {
+	// Key is the key of the condition to be matched.
+	//
+	// kubebuilder:validation:Required
+	Key string `json:"key" protobuf:"bytes,1,opt,name=key"`
+
+	// Operator is the set of operators that can be used in a scope selector requirement.
+	//
+	// kubebuilder:validation:Required
+	Operator Operator `json:"operator" protobuf:"bytes,2,opt,name=operator"`
+
+	// Value is the value of the condition to be matched.
+	//
+	// +kubebuilder:validation:Required
+	Value string `json:"value,omitempty" protobuf:"bytes,3,opt,name=value"`
+}
```

Comment:
I know this was in our design doc, but I think we may need to rethink this.

This looks a lot like a label selector or index selector, but when does a condition need to be anything other than (or more than) a single expression that evaluates to true or false?

@hiddeco will probably have an opinion on this.

Reply:
> You may need to validate an event-type and media-type for example.

But you can do that with one expression:

```
request.headers['X-Event-Type'] == 'party' && request.headers['ContentType'] == 'application/json'
```